### DISCUSSION TITLE
Add "Where" clause on xyzRow.cs

### DISCUSSION THREAD

**Discussion by osdan-madrid** [serenity-is/Serenity#6460] [Help]
Is it possible to add a "where" clause on xyzRow.cs?

I need to add:

`where KermessItemKermessId = KermesClientKermessId`

<img width="527" alt="image" src="https://user-images.githubusercontent.com/42551335/179683062-638719cf-72cb-4739-98ab-2f9a8921273b.png">

Thanks in advance :)

**Comment by edwardch** [comment]
Try to add to your list handler:

```cs
protected override void ApplyFilters(SqlQuery query)
{
    base.ApplyFilters(query);
    query.Where(fld.KermessItemKermessId == fld.KermesClientKermessId); // not sure if the equality will work, but try
}
```

Or at the top of KermessItemsDetailsRow.cs, above `public sealed class ...` 
Add `[InnerJoin()]`

You will need to add some parameters to the inner join.

**Reply by stixoffire** [reply]
This is good , the fld references MyRow.Fields ; at the top of the XYZ Repository class use using fld = MyRow.Fields;

**Comment by stixoffire** [comment]
My question would be is why do you want to get a single row for that back at all times ? 

I am wondering about your implementation of Database .. and what you are trying to achieve. If it is always that row - the data is the same.
